pulsar-client/src/main/java/org/apache/pulsar/client/impl/BinaryProtoLookupService.java

```diff
@@ -428,7 +461,9 @@ private void getTopicsUnderNamespace(InetSocketAddress socketAddress,
 
     @Override
     public void close() throws Exception {
-        // no-op
+        if (!createdLookupPinnedExecutor && lookupPinnedExecutor != null && !lookupPinnedExecutor.isShutdown()) {
```

Comment:
shouldn't this be `createdLookupPinnedExecutor` instead of `!createdLookupPinnedExecutor`?

Reply:
Done.